ruleset/decoders/0310-ssh_decoders.xml

```diff
@@ -226,12 +226,13 @@
   <order>user, srcip, srcport</order>
 </decoder>
 
-<!-- Dissconected from user -->
+<!-- Disconnected from user -->
 <!-- 2020-03-24 08:38:47.230409-0700  localhost sshd[2531]: Disconnected from user user 172.18.1.100 port 43042 -->
 <!-- 2020-03-24 08:38:47.230409-0700  localhost sshd[2531]: Disconnected from invalid user root 172.18.1.100 port 43042 -->
+<!-- 2020-03-24 08:38:47.230409-0700  localhost sshd[2531]: Disconnecting invalid user root 172.18.1.100 port 43042 -->
 <decoder name="sshd-disconnect">
   <parent>sshd</parent>
-  <prematch type="pcre2">Disconnected from (invalid )?user </prematch>
+  <prematch type="pcre2">Disconnect(ed|ing) (from )?(invalid )?user </prematch>
```

Comment:
It may be considered to eliminate the capturing of the fields in the `prematch`, because it does not seek to capture information, it only allows that text to be optional. In addition, no memory will be reserved that is not going to be used.

Regex `Disconnect(ed|ing) (from )?(invalid )?user ` 
Applied to `Disconnecting invalid user root 172.18.1.100 port 43042` 
Will match `Disconnecting invalid user `, 
And capture `Disconnecting` and `invalid` words, without using the captured fields.

```suggestion
  <prematch type="pcre2">Disconnect(?:ed|ing) (?:from )?(?:invalid )?user </prematch>
```
Also this change will avoid the analysisd issue.

Reply:
Hi @cborla !
Thanks for your review